addon/components/uni-input.js

```diff
@@ -0,0 +1,106 @@
+import Ember from 'ember';
+import layout from '../templates/components/uni-input';
+
+const { Component, computed, isEmpty } = Ember;
+const SUCCESS_CLASS = 'uni-input--success';
+const ERROR_CLASS = 'uni-input--error';
+const WARNING_CLASS = 'uni-input--warning';
+const INPUT_TYPE_NUMBER = 'number';
+const INPUT_TYPE_EMAIL = 'email';
+
+export default Component.extend({
+  layout,
+
+  classNames: ['uni-input-tooltip-container'],
+  defaultClass: ['uni-input uni-input--bordered uni-input--small'],
+  customClass: '',
+  value: null,
+  type: 'text',
+  name: '',
+  placeholder: '',
+  maxLength: null,
+  tooltipMessage: null,
+  showTooltip: false,
+  useDefaultErrorValidation: true,
```

Comment:
In my opinion, it would be simpler to have a callback called ```validateInput() {}``` that the parent passes on to the uni-input component and that checks the validity of the input. If no callback is passed, uni-input assumes the default behaviour